### PR TITLE
Refactor Game Modes

### PR DESCRIPTION
**Pull request by Zumorica**
## About the PR 
Depends on https://github.com/space-wizards/RobustToolbox/pull/2355
Functionality unchanged, except for implementation details. It compiles, and seems to work fine locally.
This would be a great candidate for a testmerge since testing locally is painful, but of course we don't have those yet...
GamePreset and GameRule have been completely revamped.
- GamePreset is now a prototype specifying game rules to apply, gamemode title and description, whether it shows in votes...
- GameRule is now an abstract class inheriting EntitySystem. It simply adds some syntax sugar for creating game rules.
- Added many new events to GameTicker stuff, which allow you to hook into player spawning for all your evil mischief needs, etc.

This should make *dynamic* and other gamemodes easier to code in the future.

### CODE REVIEW

File: Content.Server/GameTicking/Rules/SuspicionRuleSystem.cs [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+    {
+        string text;
+
+        switch (victory)
+        {
+            case Victory.Innocents:
+                text = Loc.GetString("rule-suspicion-end-round-innocents-victory");
+                break;
+            case Victory.Traitors:
+                text = Loc.GetString("rule-suspicion-end-round-trators-victory");
```
Comment (mirrorcult) [inline Content.Server/GameTicking/Rules/SuspicionRuleSystem.cs:307]: misspelled?

File: Content.Server/Suspicion/SuspicionRoleComponent.cs [hunk 2]
```diff
@@ -44,17 +43,17 @@ public class SuspicionRoleComponent : SharedSuspicionRoleComponent, IExamine
 
                 Dirty();
 
-                var suspicionRoleSystem = EntitySystem.Get<SuspicionRoleSystem>();
+                var sus = EntitySystem.Get<SuspicionRuleSystem>();
```
Comment (mirrorcult) [inline Content.Server/Suspicion/SuspicionRoleComponent.cs:46]: based

File: Resources/Prototypes/game_presets.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  rules:
+    - DeathMatch
+
+- type: gamePreset
+  id: TraitorDeathMatch
+  alias:
+    - traitordm
+    - traitordeathmatch
+  name: traitor-death-match-title
+  description: traitor-death-match-description
```
Comment (mirrorcult) [inline Resources/Prototypes/game_presets.yml:57]: this is the only one thats localized?

**Review by mirrorcult — APPROVED** [review]

**Review by moonheart08 — APPROVED** [review]
Nothing to add over mirror's review so simply approving.

File: Content.Server/GameTicking/Rules/SuspicionRuleSystem.cs (hunk 1 above)
Comment (Zumorica) [reply Content.Server/GameTicking/Rules/SuspicionRuleSystem.cs]: Seems this has always been misspelled, fixed now.

File: Resources/Prototypes/game_presets.yml (hunk 3 above)
Comment (Zumorica) [reply Resources/Prototypes/game_presets.yml:57]: Forgot to do the rest, oops. Done now!

**Review by Zumorica — COMMENTED** [review]
I approv this is legit pls believ

**Review by PaulRitter — APPROVED** [review]
yet another blessed vera-system

**Comment by PaulRitter** [comment]
grafik
apparently has some really nasty recursion completely breaking master. reverting for now

**Pull request merged**